.github/workflows/devdocs.yml

```
@@ -0,0 +1,90 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Upload dev docs
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron:  '0 0 * * *'
+
+jobs:
+  devdocs:
+    name: Fetch and upload dev docs
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout Arrow
+        uses: actions/checkout@v2
+        with:
+          repository: apache/arrow
+          path: arrow
+
+      - name: Checkout Crossbow
+        uses: actions/checkout@v2
+        with:
+          repository: ursacomputing/crossbow
+          path: crossbow
+
+      - name: Fetch Crossbow branches
+        run: |
+          cd crossbow
+          git fetch origin +refs/heads/*:refs/remotes/origin/*
```

Can we use `fetch-depth: 0` here?

https://github.com/actions/checkout/#Fetch-all-history-for-all-tags-and-branches

```suggestion
          fetch-depth: 0
```